code/modules/materials/Mat_MaterialProcs.dm

```diff
@@ -402,21 +403,24 @@ triggerOnEntered(var/atom/owner, var/atom/entering)
 			var/datum/gas/oxygen_agent_b/trace_gas = payload.get_or_add_trace_gas_by_type(/datum/gas/oxygen_agent_b)
 			payload.temperature = T0C // Greatly reduce temperature to simulate an endothermic reaction
 
-			// Itr 1: 0.125 Agent B, 10 Oxy
+			// Itr 1: 0.2 Agent B, Blegh oxy
 			// Itr 2: 0.0605 Agent B
-			// 0.1855moles/12cells=0.0155moles per cell
-			// At 0.0155 moles per cell it will take 20 iterations for reaction rate to drop below MINIMUM_REACT_QUANTITY
-			// Providing 1.3 minutes of catalyst assuming 4 sec ATMOS for a 12 cell burn chamber
-			trace_gas.moles += min(total_oxygen/1024,0.125)
-			total_oxygen -= min(trace_gas.moles*1024,total_oxygen)
-			animate_flash_color_fill_inherit(location,"#FF0000",4, 2 SECONDS)
+			// (This math is old and I dont know how to do the math to correct it, take this with a grain of salt) 0.1855moles/12cells=0.0155moles per cell
+			// (Same for this) At 0.0155 moles per cell it will take 100 iterations for reaction rate to drop below MINIMUM_REACT_QUANTITY
+			// Providing 6.5 minutes of catalyst assuming 4 sec ATMOS for a 12 cell burn chamber
```

Comment:
Assuming my math is correct... 100 iterations to deplete generating 19.099 Agent B and 100 Oxygen and will take 110 iterations to hit minimum react on reaction_rate() (7.33 minutes)

Reply:
Updated comment accordingly